protostar/commands/init/project_creator/new_project_creator.py

```diff
@@ -52,12 +66,85 @@ def _gather_input(self) -> "NewProjectCreator.NewProjectConfig":
 
     def _create_project(
         self,
-        user_input: "NewProjectCreator.NewProjectConfig",
+        project_config: "NewProjectCreator.NewProjectConfig",
         cairo_version: CairoVersion,
     ) -> None:
         output_dir_path = self._output_dir_path
-        project_root_path = output_dir_path / user_input.project_dirname
-        self.copy_template(cairo_version, project_root_path)
-        self.save_protostar_toml(
-            project_root_path=project_root_path, cairo_version=cairo_version
+        project_root_path = output_dir_path / project_config.project_dirname
+        project_name = project_config.project_dirname
+
+        self.copy_and_replace_template(
+            cairo_version=cairo_version,
+            project_root_path=project_root_path,
+            project_name=project_name,
+        )
+        self._save_protostar_toml(
+            project_root_path=project_root_path,
+            configuration_model=self._make_project_config(cairo_version=cairo_version),
+        )
+
+    def copy_and_replace_template(
```

Comment:
[nit]: `create_from_template` perhaps?

Reply:
How about `project_directory_from_template`?

Reply:
proposed name does not really describe what the method does

Reply:
Let's combine these ideas into `create_project_directory_from_template`. It's the most descriptive one.